### PR TITLE
Add pandoc support back to rust script

### PR DESCRIPTION
**Pull request by ysndr**
Addressing #329 

Pandoc support for descriptions was removed in #320 and now comes back to the rewrite

### CODE REVIEW

**Comment by jtojnar** [comment]
The [nixpkgs PR](https://github.com/NixOS/nixpkgs/pull/130047) has been merged.

**Comment by ysndr** [comment]
Great, thanks for the notice

**Comment by ysndr** [comment]
Internal links show as `???` still
Styling runs through pandoc with corresponding performance hit

**Comment by jtojnar** [comment]
Yes, we did not fix the question marks yet. In the NixOS manual, we convert them [back to `xref` tags](https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/doc/replace-xrefs-by-empty-links.py) and let DocBook resolve the chapter titles.

I think it should be possible to use manual’s olink database to get this information:

```patch
--- a/nixos/doc/manual/default.nix
+++ b/nixos/doc/manual/default.nix
@@ -159,7 +159,7 @@ let
     '';
 
 in rec {
-  inherit generatedSources;
+  inherit generatedSources olinkDB;
 
   inherit (optionsDoc) optionsJSON optionsXML optionsDocBook;
 
--- a/nixos/release.nix
+++ b/nixos/release.nix
@@ -132,6 +132,7 @@ in rec {
   manualEpub = (buildFromConfig ({ ... }: { }) (config: config.system.build.manual.manualEpub));
   manpages = buildFromConfig ({ ... }: { }) (config: config.system.build.manual.manpages);
   manualGeneratedSources = buildFromConfig ({ ... }: { }) (config: config.system.build.manual.generatedSources);
+  manualOlinkDB = buildFromConfig ({ ... }: { }) (config: config.system.build.manual.olinkDB);
   options = (buildFromConfig ({ ... }: { }) (config: config.system.build.manual.optionsJSON)).x86_64-linux;
 
 
```

**Comment by ysndr** [comment]
seems like pandoc hasnt implemented olinks support yet..

> [ ] olink - A link that addresses its target indirectly, through an entity

https://github.com/jgm/pandoc/blob/37e30560ad546055d17f56df468cd35976b3218a/src/Text/Pandoc/Readers/DocBook.hs#L287

**Comment by ysndr** [comment]
Is this at least the same state as  prior to the rewrite?

**Comment by ncfavier** [comment]
Yes, see #303.

**Comment by ysndr** [comment]
Ok, I'd say we should restore that state first.
The further issue can be handled in a separate PR

**Comment by jtojnar** [comment]
`olink`s are not used in the manual anyway. But the olink database created by docbook-xsl can be used to resolve `xref`s as well. We would need to parse the olinkDB and replace the link texts in a lua script similar to https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/doc/replace-xrefs-by-empty-links.py.

**Comment by ysndr** [comment]
Ah I see
I think we can do that rather easily indeed, although I tried not to involve any xml parsing yet

File: .github/workflows/cron-nixpkgs.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
 
   schedule:
   - cron: '0 * * * *'
+  pull_request:
```
Comment (ncfavier) [inline .github/workflows/cron-nixpkgs.yml:7]: Should this be removed?

**Review by ncfavier — APPROVED** [review]

File: .github/workflows/cron-nixpkgs.yml (hunk 1 above)
Comment (ysndr) [reply .github/workflows/cron-nixpkgs.yml:7]: Yes
I should think of a better way to do this debugging things

**Pull request merged**